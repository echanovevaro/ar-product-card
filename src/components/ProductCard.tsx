import React from 'react';
import { createContext } from 'react';
import { useProduct } from '../hooks/useProduct';
import styles from '../styles/styles.module.css';
import { ReactElement } from 'react';

import {
  IntialValues,
  OnChangeArgs,
  Product,
  ProductCardHandlers,
  PropsProductContext,
} from '../interfaces/interfaces';

export const ProductContext = createContext({} as PropsProductContext);

const { Provider } = ProductContext;

export interface Props {
  product: Product;
  // children?: ReactElement | ReactElement[];
  children: (args: ProductCardHandlers) => ReactElement | ReactElement[];
  className?: string;
  style?: React.CSSProperties;
  onChange?: (arg: OnChangeArgs) => void;
  value?: number;
  intialValues: IntialValues;
}

export const ProductCard = ({
  children,
  product,
  className,
  style,
  onChange,
  value,
  intialValues,
}: Props) => {
  const {
    counter,
    increaseBy,
    maxCount,
    isMaxCountReached,
    reset,
  } = useProduct({
    onChange,
    product,
    value,
    intialValues,
  });
  // console.log(styles);
  return (
    <Provider value={{ counter, increaseBy, product, maxCount }}>
      <div className={`${styles.productCard} ${className}`} style={style}>
        {children({
          count: counter,
          isMaxCountReached,
          maxCount,
          product,

          increaseBy,
          reset,
        })}
      </div>
    </Provider>
  );
};
