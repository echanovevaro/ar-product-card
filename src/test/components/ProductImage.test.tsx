import React from 'react';

import TestRenderer from 'react-test-renderer';

import { ProductCard, ProductImage } from '../../components';
import { product2 } from '../data/product.';

describe('ProductImage', () => {
  test('Debe de mostrar el componente correctamente con la imagen personalizada', () => {
    const testRenderer = TestRenderer.create(
      <ProductImage img="https://googleapis.jpg/" />
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  test('Debe de mostrar el componente correctamente con la imagen del producto', () => {
    const testRenderer = TestRenderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
