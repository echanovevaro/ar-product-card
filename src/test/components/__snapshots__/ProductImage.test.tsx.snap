// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente correctamente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="product"
    className="productImg  undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage Debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="product"
  className="productImg  undefined"
  src="https://googleapis.jpg/"
/>
`;
